test/unit/formatting_stack/strategies/impl/git_diff.clj

```diff
@@ -30,6 +30,17 @@
     "B\tfoo.clj" "foo.clj"
     "foo.cljB\t" "foo.cljB\t"
 
+    ;; merge conflicts are replaced
+    "DU\t"        ""
+    "DU\tfoo.clj" "foo.clj"
+    "foo.cljDU\t" "foo.cljDU\t"
+
+    ;; `R` is replaced including optional previous path
+    "R001\tfoo.clj"          "foo.clj"
```

Comment:
What's the output for `"R001R001\tfoo.clj"`?

I'd be wary in the case `R001R001` was trimmed. b/c it's not a real git input, and we shouldn't create confusing outputs (which would hinder future debugging, etc)

Reply:
> I'd be wary in the case R001R001 was trimmed. 

That's the case because everthing between `R` and `\t` is stripped. this is intended as it contains a random int and it's previous filepath.

The whole regex is a bit brittle as is, but i cant really find a more proper git-tool to find the information we're looking for anyway.

Reply:
Mmmm, I cannot have enough confidence in the proposed changes atm. A single `\t` shouldn't cause f-s to eat arbitrary inputs.

Today's bug shows that there are always "unknown unknowns", in which deterministic/delimited behavior helps.

> The whole regex is a bit brittle as is

No particular need for the thing to be a regex (or at least a _single_ regex). This is also fine: `(-> s (string/replace regex-or-str) (string/replace regex-or-str) (string/replace regex-or-str))` (point being: N calls can be acceptable vs. a contrived regex)

Needless to say, existing tests increase our confidence in swapping impls :)

Reply:
>  A single \t shouldn't cause f-s to eat arbitrary inputs.

agreed, it works because coincidence rather than design.

Having multiple regexes without a clear doc on all possible outputs also seems problematic. Am working on a suggestion using `diff-filter` which should be more stable.